api/models/Campaign.js

```diff
@@ -31,8 +35,51 @@ const schema = new mongoose.Schema({
   msg_menu_signedup_gambit: String,
   msg_no_photo_sent: String,
 
+  // Mobile Commons Specific Fields.
+  mobilecommons_groups: {
+    doing: Number,
+    completed: Number,
+  },
+
 });
 
+/**
+ * For a given campaign, update its mobile commons group id,
+ * based on the mobile commons response.
+ * @param {Campaign} campaign    Campaign model to update.
+ * @param {string} status        Either completed or doing.
+ * @param {string} groupResponse XML string from mobile commons API.
+ */
+function setMobileCommonsGroup(campaign, status, groupResponse) {
+  const scope = campaign;
+  const parsedGroup = JSON.parse(parser.toJson(groupResponse));
```

Comment:
My 0.02 is it would be a lot cleaner to parse the JSON in `lib/mobilecommons.js` -- especially since we'll want to parse there too for a `getGroups` function last resort in https://github.com/DoSomething/gambit/issues/673#issuecomment-256251277